### PR TITLE
Update Habitat plan to correctly build

### PR DESCRIPTION
**Pull request by elliott-davis**
### Description

Updates habitat plan to build chef 13 correctly. Also adds config settings for data-collector

### Check List

- [ ] New functionality includes tests
- [ ] All tests pass
***There are no integration tests to validate this***
- [x] RELEASE\_NOTES.md, has been updated if required (not required for bugfixes, required for API changes)
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: habitat/default.toml [hunk 1]
```diff
@@ -1,11 +1,28 @@
-# The Chef run interval in seconds
-interval = 10 
+[chef-client]
```
Comment (smith) [inline habitat/default.toml]: Should we be using underscores here? Would like to get consistency about this across core/chef plans. Not a blocker, but the other variables use underscores in this file.

Comment (elliott-davis) [inline habitat/default.toml]: the more I think about it - I don't really like having that chef-client table there at all. I think I'll remove it unless you have any objections. After looking through the docs there are only a couple of nested keys so it's sort of redundant to have a chef-client table in the chef-client package.

File: habitat/config/client.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+cookbook_path "{{pkg.svc_data_path}}/{{cfg.cookbook_path}}"
+data_bag_path "{{pkg.svc_data_path}}/{{cfg.data_bag_path}}"
+environment_path "{{pkg.svc_data_path}}/{{cfg.environment_path}}"
+file_backup_path "{{pkg.svc_data_path}}/{{cfg.file_backup_path}}"
+node_path "{{pkg.svc_data_path}}/{{cfg.node_path}}"
+role_path "{{pkg.svc_data_path}}/{{cfg.role_path}}"
+pid_file "{{pkg.svc_data_path}}/{{cfg.pid_file}}"
```
Comment (tduffield) [inline habitat/config/client.rb]: nit: this feels like it should go in `svc_var_path`

File: habitat/hooks/run [hunk 3]
```diff
@@ -1,8 +1,6 @@
...
-export GEM_HOME="{{pkg.path}}/ruby/2.3.0"
-export GEM_PATH="$(hab pkg path core/ruby)/lib/ruby/gems/2.3.0:$(hab pkg path core/bundler):$GEM_HOME"
-export APPBUNDLER_ALLOW_RVM=true
-
 exec 2>&1
-exec chef-solo --fork -i {{cfg.interval}} -s {{cfg.splay}} -l {{cfg.log_level}} -o {{cfg.run_list}}
 
+export SSL_CERT_FILE="{{pkgPathFor "core/cacerts"}}/ssl/cert.pem"
+
+exec chef-solo --fork -c {{pkg.svc_config_path}}/client.rb
```
Comment (tduffield) [inline habitat/hooks/run]: I (think) this might want to be `--no-fork`?

Comment (rhass) [inline habitat/hooks/run]: We can't use `--no-fork` as it is required in Chef 12 and later. We can however remove the exec since it is redundant.

Comment (adamhjk) [reply habitat/hooks/run]: No it’s not. It means the pid doesn’t change.

**Pull request merged**